### PR TITLE
Integrate TRC

### PR DESCRIPTION
**Pull request by shgandhi**
This PR moves refactored TRC to open-source. This PR modifies the CMakeLists.txt to allow the move.

### CODE REVIEW

File: client/thin-replica-client/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
+add_library(thin_replica_client SHARED
+  "src/thin_replica_client.cpp"
+  "src/trace_contexts.cpp"
+  "src/trs_connection.cpp"
+  "src/crypto_utils.cpp"
+  "src/trc_hash.cpp")
+target_include_directories(thin_replica_client PUBLIC include)
+target_compile_definitions(thin_replica_client PUBLIC USE_LOG4CPP)
+target_include_directories(thin_replica_client PUBLIC
```
Comment (robem) [inline client/thin-replica-client/CMakeLists.txt:18]: Does this need to be public? If someone depends on the secrets manager then they should include it directly.

File: client/thin-replica-client/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+target_link_libraries(thin_replica_client
+  log4cplus
+  opentracing
+  util
+  cryptopp
+  thin-replica-proto
+)
+
+install(TARGETS thin_replica_client DESTINATION lib)
+install(FILES include/client/thin-replica-client/thin_replica_client.hpp include/client/thin-replica-client/health_status.hpp include/client/thin-replica-client/update.hpp DESTINATION "include/thin_replica_client")
```
Comment (robem) [inline client/thin-replica-client/CMakeLists.txt:29]: GitHub marks a missing new line at the end.

File: thin-replica-server/proto/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+...
+")
+
+protobuf_generate_cpp(PROTO_SRCS PROTO_HDRS ${CMAKE_CURRENT_BINARY_DIR}
+  thin_replica.proto
+)
+grpc_generate_cpp(GRPC_SRCS GRPC_HDRS ${CMAKE_CURRENT_BINARY_DIR}
+  thin_replica.proto
+)
+message(STATUS "gRPC/protobuf generated - see " ${CMAKE_CURRENT_BINARY_DIR})
```
Comment (robem) [inline thin-replica-server/proto/CMakeLists.txt:23]: Maybe be more specific: "thin replica gRPC protocol" or so.

File: thin-replica-server/proto/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+add_library(thin-replica-proto SHARED ${PROTO_SRCS} ${GRPC_SRCS})
```
Comment (robem) [inline thin-replica-server/proto/CMakeLists.txt:25]: Can we make this static?

File: thin-replica-server/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,3 @@
+project("Thin Replica Server")
+
+add_subdirectory("proto")
```
Comment (robem) [inline thin-replica-server/CMakeLists.txt:3]: New line again and also more below.

**Review by robem — APPROVED** [review]
Looks good to me. Thanks for the hard work 💪 
Only minor comments that can be addressed as follow-ups if you agree.

**Comment by andrewjstone** [comment]
EPIC! Exciting stuff @shgandhi! 🚀 🎉

**Comment by shgandhi** [comment]
Thanks for the review @robem. I have updated the PR according to your comments.

File: client/thin-replica-client/src/trace_contexts.cpp [hunk 6]
```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 VMware, all rights reserved
+
```
Comment (upshaw-alex) [inline client/thin-replica-client/src/trace_contexts.cpp:2]: Does this file not need one of those license statements that begins "This product is licensed to you under..."?

File: client/thin-replica-client/include/client/thin-replica-client/trace_contexts.hpp [hunk 7]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2021 VMware, Inc. All rights reserved. VMware Confidential
+
```
Comment (upshaw-alex) [inline client/thin-replica-client/include/client/thin-replica-client/trace_contexts.hpp:2]: Does this file not need one of those license statements that begins "This product is licensed to you under..."?

**Pull request merged**